spec.md

```diff
@@ -570,13 +559,297 @@ schema being used.
+Broker can signal to the Platform that JSON Schema components are supported by
+using the `osb_v2:` scheme. Components are a way to allow Platforms to only
+fetch sub-schemas.
+
+The root JSON Schema document can be found by getting the JSON Schemas
+endpoint without a `:component_path`.
+
+If the `osb_v2:` scheme is used by the Service Broker, then the Platform MAY
+fetch the component JSON Schema directly using the following convention:
+
+ - Given: `osb_v2:///{:component_path}`, fetch
+   `/v2/catalog/schemas/:component_path` and then the Platform is responsible
+   to continue fetching any unknown `:component_path` URIs. Care ought to be
+   taken by the platform to not fetch duplicate definitions.
+
+A Platform MAY ignore the `osb_v2:` scheme and just fetch `/v2/catalog/schemas`
+which will contain all sub-Schema documents. This reduces the complexity of the
+Platform but increases the potential payload size of the JSON Schemas.
+
+A Platform SHOULD fetch the root Schema document for any other `$ref` URIs
+found that do not match the `osb_v2:` scheme. The Service Broker MUST provide
+the JSON Schema Object for the referenced URI in a subschema returned.
+
+Note: If the Service Broker provides a mix of `osb_v2:` in addition to other
+schemes that result in fetching both `/v2/catalog/schemas/:component_path` and
+`/v2/catalog/schemas`, the Platform will have colliding JSON Schema `$ids`. In
+this case the Platform SHOULD ignore the component paths and use the root
+schema. Service Brokers SHOULD avoid mixing `osb_v2:` and other schemes.
+
+When the osb_v2: scheme is used, the data referenced by a component path MUST
+NOT change and can only be deleted when there are no longer any references to
+it in the Service Broker's schema. Additionally, once deleted, that component
+path MUST NOT ever be used again by this Service Broker, unless the schema
+referenced is the exact same as the previously referenced schema.
+
+A note about the scheme, at the time of this writing, there were no found client
+libraries that fully support the [JSON Schema](http://json-schema.org/) spec in
+the usage of URI for `$ref`. Only URL was supported in libraries. So as a
+compromise the `osb_v2:` scheme SHOULD be followed by three forward slashes
+(`///`). This indicates the authority is implied (authority is '').
+
+
+#### $ref Usage
+
+Service Brokers MAY leverage `$ref` and external JSON Schemas only if the JSON
+Schema references can be found within the [JSON Schemas](#json-schemas) API
+endpoint. `$ref` usage is assumed to be delegation (linking), not inclusion
+(replacement). Libraries that implement `$ref` content replacement result in
+adverse behaviors.
+
+For Service Brokers that support `$ref`, they:
+
+  * MUST respond with all JSON Schema documents for `/schemas` requests that are
+    referenced in the catalog.
+  * MAY provide `$ref`'s with the `osb_v2:` scheme, then:
+    * MUST respond to `/schemas/:component_id` requests.
```

Comment:
this is the only occurrence of component_id ... I was looking for its definition (i.e. whether it is a simple identifier, or whether a path is allowed, to give broker more control).

Then I found out that elsewhere :component_path is used. Are they intended to be the same, or are they different (and the :component_id needs more clarification)?